ank/src/cli_commands/set_state.rs

```diff
@@ -117,9 +156,230 @@ impl CliCommands {
+    #[tokio::test]
+    async fn utest_process_inputs_stdin() {
+        let input = SAMPLE_CONFIG;
+        let reader = Cursor::new(input);
+        let state_object_file = "-".to_string();
+
+        let temp_obj = process_inputs(reader, &state_object_file).await.unwrap();
+
+        let value: Value = serde_yaml::from_str(SAMPLE_CONFIG).unwrap();
+        let expected_obj = Object::try_from(&value).unwrap();
+
+        assert_eq!(temp_obj, expected_obj);
+    }
+
+    // [utest->swdd~cli-supports-yaml-to-set-desired-state~1]
+    #[tokio::test]
+    async fn utest_process_inputs_file() {
+        let state_object_file = SAMPLE_CONFIG.to_owned();
+        println!("{:?}", state_object_file);
+
+        let temp_obj = process_inputs(io::empty(), &state_object_file)
+            .await
+            .unwrap();
+        println!("{:?}", temp_obj);
+
+        let value: Value = serde_yaml::from_str(SAMPLE_CONFIG).unwrap();
+        let expected_obj = Object::try_from(&value).unwrap();
+        println!("{:?}", expected_obj);
+
+        assert_eq!(temp_obj, expected_obj);
+    }
+
+    // [utest->swdd~cli-supports-yaml-to-set-desired-state~1]
+    #[tokio::test]
+    async fn utest_process_inputs_invalid_yaml() {
+        let input = "invalid yaml";
+        let reader = Cursor::new(input);
+        let state_object_file = "-".to_string();
+
+        let temp_obj = process_inputs(reader, &state_object_file).await;
+
+        assert!(temp_obj.is_ok());
+    }
+
+    // [utest->swdd~cli-provides-set-desired-state~1]
+    #[tokio::test]
+    async fn utest_set_state_ok() {
+        let update_mask = vec!["desiredState.workloads.nginx.restartPolicy".to_string()];
+        let state_object_file = SAMPLE_CONFIG.to_owned();
+
+        let workload_spec = StoredWorkloadSpec {
+            restart_policy: RestartPolicy::Always,
+            ..Default::default()
+        };
+        let updated_state = CompleteState {
+            desired_state: State {
+                workloads: HashMap::from([("nginx".to_string(), workload_spec)]),
+                ..Default::default()
+            },
+            ..Default::default()
+        };
+        let mut mock_server_connection = MockServerConnection::default();
+        mock_server_connection
+            .expect_update_state()
+            .with(eq(updated_state), eq(update_mask.clone()))
+            .return_once(|_, _| {
+                Ok(UpdateStateSuccess {
+                    added_workloads: vec![],
+                    deleted_workloads: vec![],
+                })
+            });
+
+        let mut cmd = CliCommands {
+            _response_timeout_ms: RESPONSE_TIMEOUT_MS,
+            no_wait: true,
+            server_connection: mock_server_connection,
+        };
+
+        let set_state_result = cmd.set_state(update_mask, state_object_file).await;
+        assert!(set_state_result.is_ok());
+    }
+
+    #[tokio::test]
+    async fn utest_set_state_failed() {
```

Comment:
linkage is missing: `// [utest->swdd~cli-provides-set-desired-state~1]`

In addition, the utest setup can be simplified and I suggest to use a more explicit wrong field mask for the readability:

```rust
        let wrong_update_mask = vec!["desiredState.workloads.notExistingWorkload".to_string()];
        let state_object_file = SAMPLE_CONFIG.to_owned();
        let mut mock_server_connection = MockServerConnection::default();
        mock_server_connection
            .expect_update_state()
            .return_once(|_, _| Ok(UpdateStateSuccess::default()));

        let mut cmd = CliCommands {
            _response_timeout_ms: RESPONSE_TIMEOUT_MS,
            no_wait: true,
            server_connection: mock_server_connection,
        };

        let set_state_result = cmd.set_state(wrong_update_mask, state_object_file).await;
        assert!(set_state_result.is_err());
```